LanguageFeatures/Augmentations/augmenting_functions_A04_t18.dart

```diff
@@ -2,161 +2,291 @@
+/// matches if:
+/// - It has the same number of type parameters with the same type parameter
+///   names (same identifiers) and bounds (after type annotation inheritance),
+///   if any (same types, even if they may not be written exactly the same in
+///   case one of the declarations needs to refer to a type using an import
+///   prefix).
+/// - The return type (if not omitted) is the same as the augmented
+///   declaration's return type.
+/// - It has the same number of positional and optional parameters as the
+///   augmented declaration.
+/// - It has the same set of named parameter names as the augmented declaration.
+/// - For all corresponding pairs of parameters:
+///   - They have the same type (if not omitted in the augmenting declaration).
+///   - They have the same `required` and `covariant` modifiers.
+/// - For all positional parameters:
+///   - The augmenting function's parameter name is `_`, or
+///   - The augmenting function's parameter name is the same as the name of the
+///     corresponding positional parameter in every preceding declaration that
+///     doesn't have `_` as its name.
+/// ...
+/// It's a compile-time error if:
+/// - The signature of the augmenting function does not match the signature of
+///   the augmented function.
 ///
-/// @description Checks that it is not an error if return type and  parameters
-/// of an augmentation exactly matches the original function.
+/// @description Checks that it is not an error if return type and parameters of
+/// an augmentation exactly matches the original function.
 /// @author sgrekhov22@gmail.com
 
-// SharedOptions=--enable-experiment=macros
+// SharedOptions=--enable-experiment=augmentations
 
 import '../../Utils/expect.dart';
-part 'augmenting_functions_A04_t18_lib.dart';
 
 String topLevelFunction1() => "42";
+
+augment String topLevelFunction1();
+
 Object topLevelFunction2(String s) => s;
+
+augment Object topLevelFunction2(String s);
+
 int topLevelFunction3([int i = 0]) => i;
-int? topLevelFunction4({int? i, int j = 0}) => null;
+
+augment int topLevelFunction3([int i]);
+
+int? topLevelFunction4({int? i}) => i;
+
+augment int? topLevelFunction4({int? i});
+
 T topLevelFunction5<T extends num>(T t) => t;
 
+augment T topLevelFunction5<T extends num>(T t);
+
 class C {
   static String staticMethod1() => "42";
   static Object staticMethod2(String s) => s;
   static int staticMethod3([int i = 0]) => i;
-  static int? staticMethod4({int? i, int j = 0}) => null;
+  static int? staticMethod4({int? i = 0}) => i;
   static T staticMethod5<T extends num>(T t) => t;
+
   String instanceMethod1() => "42";
   Object instanceMethod2(String s) => s;
   int instanceMethod3([int i = 0]) => i;
-  int? instanceMethod4({int? i, int j = 0}) => null;
+  int? instanceMethod4({int? i}) => i;
   T instanceMethod5<T extends num>(T t) => t;
 }
 
+augment class C {
+  augment static Object staticMethod1();
+  augment static String staticMethod2(String s);
```

Comment:
Shouldn't those return types be swapped?
```suggestion
  augment static String staticMethod1();
  augment static Object staticMethod2(String s);
```